### PR TITLE
[14.0][IMP] sale_quotation_number, sale_isolated_quotation, sale_partner_version, sale_order_type: resolve conflicts

### PR DESCRIPTION
**Pull request by hailangvn**
* Two modules `sale_quotation_number`, `sale_isolated_quotation` are very similar in their functional purpose, with a subtle difference as discussed at https://github.com/OCA/sale-workflow/issues/1658#issuecomment-885076018. This PR is to resolve conflict between the two modules to reduce test overhead and also code redundancy.
* The conflict of module `sale_partner_version` is resolved as well and depends on https://github.com/OCA/stock-logistics-workflow/pull/940

### CODE REVIEW

**Comment by rousseldenis** [comment]
@hailangvn Those modules are in a stable version. I would tend to do this kind of high refactor in v15.

**Comment by hailangvn** [comment]
Thanks @rousseldenis for your comment. The main purpose of this improvement is to reduce test overhead. In stead of 6 test jobs as in https://github.com/OCA/sale-workflow/runs/4382557420 we will have only 2 jobs as in https://github.com/OCA/sale-workflow/pull/1819/checks?check_run_id=4501777323. Let's preserve precious computing energy for greener environment.

**Comment by rousseldenis** [comment]
> Title



> Thanks @rousseldenis for your comment. The main purpose of this improvement is to reduce test overhead. In stead of 6 test jobs as in https://github.com/OCA/sale-workflow/runs/4382557420 we will have only 2 jobs as in https://github.com/OCA/sale-workflow/pull/1819/checks?check_run_id=4501777323. Let's preserve precious computing energy for greener environment.

Arf, never heard such argument...

But I see you don't remove module, so, ok for me as I see there are migration scripts.

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
And by the way, this fixes runboat builds @sbidoul

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+odoo14-addon-stock_partner_delivery_window @git+https://github.com/OCA/stock-logistics-workflow@refs/pull/940/head#subdirectory=setup/stock_partner_delivery_window
```
Comment (sbidoul) [inline test-requirements.txt]: A good habit is to keep PRs in Draft state while there are dependencies on unmerged addons. This avoids merging by accident.

**Comment by hailangvn** [comment]
Thanks @rousseldenis for your support. Thanks @sbidoul for your advice. This PR is now in draft state.

**Comment by rousseldenis** [comment]
@hailangvn https://github.com/OCA/stock-logistics-workflow/pull/940 merged

**Comment by hailangvn** [comment]
Thanks @rousseldenis, this PR has been updated and rebased.

**Comment by rousseldenis** [comment]
@OCA/crm-sales-marketing-maintainers What is your advice on this ?

It allows to fix a recurrent problem on 14.0 branch about conflicting modules making them dependent from each other.

It seems ok for me. Wanted to merge with major version update

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Review by bealdav — APPROVED** [review]

**Comment by bealdav** [comment]
Finally @sbidoul do you think it's ok now ? Thanks

**Comment by sbidoul** [comment]
> Finally @sbidoul do you think it's ok now ? Thanks

I am not familiar with these modules and I don't have time to review closer. But everything is green :)

**Comment by bealdav** [comment]
ok we'll let @rousseldenis operate on it

**Pull request merged**